packages/synapse-react-client/src/components/UserOrTeamBadge/UserOrTeamBadge.tsx

```diff
@@ -29,23 +28,14 @@ export default function UserOrTeamBadge(props: UserOrTeamBadgeProps) {
     principalId = providedUserGroupHeader?.ownerId
   }
 
-  const { accessToken } = useSynapseContext()
-  const [userGroupHeader, setUserGroupHeader] = useState<
-    UserGroupHeader | undefined
-  >(providedUserGroupHeader)
-
   const { data: fetchedUserGroupHeader } = useGetUserGroupHeader(
     (principalId ?? '').toString(),
     {
       enabled: !providedUserGroupHeader,
     },
   )
 
-  useEffect(() => {
-    if (principalId && userGroupHeader == undefined && fetchedUserGroupHeader) {
-      setUserGroupHeader(fetchedUserGroupHeader)
-    }
-  }, [accessToken, principalId, userGroupHeader, fetchedUserGroupHeader])
```

Comment:
In particular, this effect would never re-run when the `principalId` changed, so the data remained stale.